lightning/src/chain/keysinterface.rs

```diff
@@ -133,7 +133,7 @@ pub trait KeysInterface: Send + Sync {
 /// (TODO: We shouldn't require that, and should have an API to get them at deser time, due mostly
 /// to the possibility of reentrancy issues by calling the user's code during our deserialization
 /// routine).
-pub trait ChannelKeys : Send {
+pub trait ChannelKeys : Send+Clone+PartialEq {
```

Comment:
Oof. Clone sucks, especially since the enforcement then cant be in our own ChannelKeys but has to be some global hashmap tracking. I suppose its largely something that can only be resolved with #387, so if you just add a comment noting that we need to remove this ASAP, maybe that's OK. Also, can we drop the PartialEq and have the ChannelMonitor PartialEq compare pubkeys()?

Reply:
Done, on the Storage enum